testng/src/main/java/io/cucumber/testng/Pickle.java

```diff
@@ -2,11 +2,17 @@
 
 import org.apiguardian.api.API;
 
+import java.net.URI;
+import java.util.List;
+
+import io.cucumber.core.gherkin.Location;
+import io.cucumber.core.gherkin.Step;
+
 /**
  * Wraps CucumberPickle to avoid exposing it as part of the public api.
  */
 @API(status = API.Status.STABLE)
-public final class Pickle {
+public final class Pickle implements io.cucumber.core.gherkin.Pickle {
```

Comment:
By implementing ` io.cucumber.core.gherkin.Pickle` we're exposing internal parts of the API to external users.  I don't think this interface here is needed to add all the methods so we can remove this.

Reply:
Additionally I don't think all information should be exposed. For now I'd like to limit this to `name`, `tags`, `uri` and `scenarioLine`.

Reply:
I updated PR accordingly.